### ISSUE TITLE
How to define two optimizer for different layers in one ModelDesc

### ISSUE THREAD

**Issue by yogurfrul** [tensorpack/tensorpack#842]
I want to implement [C3D-tensorflow](https://github.com/hx173149/C3D-tensorflow) to tensorpack
while the model define two AdamOptimizer with different learning rate for different variable
```
opt_stable = tf.train.AdamOptimizer(1e-4)
opt_finetuning = tf.train.AdamOptimizer(1e-3)
grads1 = opt_stable.compute_gradients(loss, varlist1)
grads2 = opt_finetuning.compute_gradients(loss, varlist2)
apply_gradient_op1 = opt_stable.apply_gradients(grads1)
apply_gradient_op2 = opt_finetuning.apply_gradients(grads2, global_step=global_step)
train_op = tf.group(apply_gradient_op1, apply_gradient_op2)
```
I can only find some way to apply different lr for different step, not for different layers
```
class C3DModel(ModelDesc):
       xxx

        def optimizer(self):
            return tf.train.AdamOptimizer(1e-4)
```
How can I implement the model to tensorpack rightly

**Comment by ppwwyyxx** [comment]
Use http://tensorpack.readthedocs.io/modules/tfutils.html#tensorpack.tfutils.optimizer.apply_grad_processors with http://tensorpack.readthedocs.io/modules/tfutils.html#tensorpack.tfutils.gradproc.ScaleGradient.

More generally, you can implement a `tf.train.Optimizer` yourself by inheriting an existing optimizer such as Adam and making some modifications.

**Comment by yogurfrul** [comment]
varlist2 is wout and bout in fact ,so I use apply_grad_processors and ScaleGradient as below:

```
 def optimizer(self):
        opt = tf.train.AdamOptimizer(1e-4)
        return optimizer.apply_grad_processors(
            opt, [gradproc.ScaleGradient(
                [('.*/wout', 10), ('.*/bout', 10)],log=True)])
```

I found this print in log
```
[0727 06:38:03 @gradproc.py:251] Apply lr multiplier 10 for var_name/wout
[0727 06:38:03 @gradproc.py:251] Apply lr multiplier 10 for var_name/bout
```
I think it works~
thx

**Comment by yogurfrul** [comment]
Can I join two different optimizer to one in ModelDesc?
opt1 = tf.train.AdamOptimizer(1e-4)
opt2 = tf.train.GradientDescentOptimizer(1e-3)

**Comment by ppwwyyxx** [comment]
That when you need to write an optimizer yourself. Just call `opt1.apply_gradients` or `opt2.apply_gradients` in your own `apply_gradients`.

**Comment by yogurfrul** [comment]
I see~
thx
